components/tikv_util/src/metrics/threads_dummy.rs

```diff
@@ -36,3 +36,9 @@ impl ThreadInfoStatistics {
         HashMap::default()
     }
 }
+
+impl Default for ThreadInfoStatistics {
```

Comment:
We allow clippy::new_without_default explictly.

Reply:
OK, so we can remove all the Default method is this pr?

Reply:
Are there any command to allow clippy::new_without_default in clippy?

Reply:
It's at https://github.com/tikv/tikv/blob/master/scripts/clippy#L20

Reply:
So all the lints in this script is allowed?

Reply:
`unnecessary_wraps` and `needless_range_loop` can be removed, they are not checked probably due to too many changes need to be taken.

Reply:
I think it may be better to fix the clippy errors by package. And I have fixed the errors in the `tikv_util` package.